File: University Student Rating/Code/androidclient/UniversityStudentRating/app/src/main/java/com/example/arturrinkis/universitystudentrating/ServerServiceAPI/Interfaces/Services/IStatisticService.java
package com.example.arturrinkis.universitystudentrating.ServerServiceAPI.Interfaces.Services;

import com.example.arturrinkis.universitystudentrating.DTO.Discipline;
import com.example.arturrinkis.universitystudentrating.DTO.DisciplineBranch;
import com.example.arturrinkis.universitystudentrating.DTO.IndividualRating;
import com.example.arturrinkis.universitystudentrating.DTO.ProfileRating;
import com.example.arturrinkis.universitystudentrating.DTO.Rating;
import com.example.arturrinkis.universitystudentrating.DTO.RatingType;

import java.util.ArrayList;

public interface IStatisticService {
    ArrayList<Rating> getRatingsForProfessorByDiscipline(int professorId, int disciplineId, int skipCount, int takeCount);
    ArrayList<DisciplineBranch> getDisciplineBranches();
    ArrayList<Discipline> getDisciplinesByBranch(int branchId);
    ProfileRating getProfileRating(int studentId, int courseId);
    IndividualRating getIndividualRating(int studentId, int disciplineId);
    ArrayList<IndividualRating> getAverageClassTopStudentsBy(int universityId, int courseId, int disciplineId, int topCount);
    ArrayList<IndividualRating> getOverallTopStudentsBy(int universityId, int courseId, int disciplineId, int topCount);
    ArrayList<IndividualRating> getOlympiadsTopStudentsBy(int universityId, int courseId, int disciplineId, int topCount);
    RatingType getRatingType(int universityId,  int disciplineId);
    double getAverageRatingForProfessor(int professorId);
    boolean setRating(Rating rating);
    boolean deleteRatingByIds(ArrayList<Integer> ratingIds);
}

File: University Student Rating/Code/server/UniversityStudentRatingService/ServiceLogicLayer/Interfaces/IStatisticService.cs
using DataAccessLayer.DbContext;
using ServiceLogicLayer.DTO;
using ServiceLogicLayer.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLogicLayer.Interfaces
{
    public interface IStatisticService
    {
        IEnumerable<DisciplineBranch> GetDisciplineBranches();
        IEnumerable<Discipline> GetDisciplinesByBranch(int branchId);
        IEnumerable<Rating> GetRatingBy(int studentId, int ratingDivisionId, int branchId);
        IEnumerable<Rating> GetRatingBy(int studentId, int ratingDivisionId);
        IEnumerable<Rating> GetRatingsForProfessorByDiscipline(int professorId, int disciplineId, int skipCount, int takeCount, IUserService userService);

        double GetAverageRatingByDivision(int studentId, int ratingDivisionId);
        double GetAverageRatingByBranchDivision(int studentId, int branchId, int ratingDivisionId);
        double GetAverageRatingByBranchDisciplineDivision(int studentId, int branchId, int disciplineId, int ratingDivisionId);
        double GetAverageRatingForProfessor(int professorId);

        int GetTotalRating(int studentId);
        int GetTotalRatingByBranch(int studentId, int branchId);
        int GetTotalRatingByBranchDiscipline(int studentId, int branchId, int disciplineId);
        int GetTotalRatingByDivision(int studentId, int divisionId);
        int GetTotalRatingByBranchDivision(int studentId, int branchId, int divisionId);
        int GetTotalRatingByBranchDisciplineDivision(int studentId, int branchId, int disciplineId, int divisionId);

        int GetMaxRating(int courseId);
        int GetMaxOlympiadsRating(int courseId);
        int GetMaxRatingByBranch(int courseId, int branchId);
        
        ProfileRatingDTO GetProfileRating(int studentId, int courseId);
        IndividualRatingDTO GetIndividualRating(int studentId, int disciplineId, string academicYear);
        IndividualRatingDTO GetIndividualRating(int studentId, int disciplineId);
        List<IndividualRatingDTO> GetAverageClassTopStudentsBy(int universityId, int courseId, int disciplineId, int topCount);
        List<IndividualRatingDTO> GetOverallTopStudentsBy(int universityId, int courseId, int disciplineId, int topCount);
        List<IndividualRatingDTO> GetOlympiadsTopStudentsBy(int universityId, int courseId, int disciplineId, int topCount);
        RatingType GetRatingType(int universityId, int disciplineId);
        bool SetRating(Rating model);
        bool DeleteRatingByIds(List<int> ratingIds);
    }
}